node/remotenode.go

```diff
@@ -8,14 +8,15 @@ import (
 
 	"github.com/golang/protobuf/proto"
 	"github.com/nknorg/nkn/v2/pb"
+	"github.com/nknorg/nkn/v2/util/log"
 	nnetnode "github.com/nknorg/nnet/node"
 )
 
 type RemoteNode struct {
 	*Node
-	localNode *LocalNode
-	nnetNode  *nnetnode.RemoteNode
-	sharedKey *[sharedKeySize]byte
+	localNode ILocalNode
+	NnetNode  *nnetnode.RemoteNode
+	SharedKey *[SharedKeySize]byte
```

Comment:
Don't expose sharedKey unless absolutely necessary

Reply:
yes. I moved encryptMessage and decryptMessage function into node package. so remoteNode no need expose its sharedKey.